tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/lock/LocksWithOnFailTest.java

```diff
@@ -35,11 +35,10 @@
  * @since 0.961.0
  */
 public class LocksWithOnFailTest {
+    CompileResult compileResult = BCompileUtil.compile("test-src/lock/lock-on-fail.bal");
```

Comment:
Shall we add this to a ` @BeforeClass`? Otherwise, this compilation is going to run every time. (https://github.com/ballerina-platform/ballerina-lang/pull/36394)

We also need to add 
```java
    @AfterClass
    public void tearDown() {
        compileResult = null;
    }
```
Otherwise may contribute to OOM